### PR TITLE
Enable CLASS and SET support in codegen-py

### PR DESCRIPTION
**Pull request by jpbusch**
Generate CPM msgs and conversion by `codegen-py` instead of `codegen-rust`. All files are now generated with python again instead of rust (also mentioned in readme 622996ea14fb2a285876750684dc02d7e3b0771c).

### Breaking Changes

<details>
<summary>etsi_its_cpm_ts_msgs</summary>

- `CollectivePerceptionMessage.msg`
    ```diff
    - int64 HEADER_MIN = 0
    ```
- `CpmContainers.msg`: *removed*
- `CorrelationColumn.msg`
    ```diff
    - int64 MIN_SIZE = 1
    - int64 MAX_SIZE = 13
    + uint8 MIN_SIZE = 1
    + uint8 MAX_SIZE = 13
    ```
- `MatrixIncludedComponents.msg`
    ```diff
    - int64 MIN_SIZE_BITS = 13
    - int64 MAX_SIZE_BITS = 13
    + uint8 SIZE_BITS = 13
    ```
- `MessageRateHz.msg`
    ```diff
    - uint8 EXPONENT_MIN = 0
    - uint8 EXPONENT_MAX = 2
    + int8 EXPONENT_MIN = -5
    + int8 EXPONENT_MAX = 2
    ```
- `PerceivedObjectsIds.msg`
    ```diff
    - int64 MIN_SIZE = 0
    - int64 MAX_SIZE = 255
    + uint8 MIN_SIZE = 0
    + uint8 MAX_SIZE = 255
    ```
- `PerceivedObjects.msg`
    ```diff
    - int64 MIN_SIZE = 0
    - int64 MAX_SIZE = 255
    + uint8 MIN_SIZE = 0
    + uint8 MAX_SIZE = 255
    ```
- `PerceptionRegionContainer.msg`
    ```diff
    - int64 MIN_SIZE = 1
    - int64 MAX_SIZE = 256
    + uint16 MIN_SIZE = 1
    + uint16 MAX_SIZE = 256
    ```
- `PolygonalShape.msg`
    ```diff
    - int64 POLYGON_MIN_SIZE = 3
    - int64 POLYGON_MAX_SIZE = 16
    + uint8 POLYGON_MIN_SIZE = 3
    + uint8 POLYGON_MAX_SIZE = 16
    ```
- `RadialShapesList.msg`
    ```diff
    - int64 MIN_SIZE = 1
    - int64 MAX_SIZE = 16
    + uint8 MIN_SIZE = 1
    + uint8 MAX_SIZE = 16
    ```
- `SensorInformationContainer.msg`
    ```diff
    - int64 MIN_SIZE = 1
    - int64 MAX_SIZE = 128
    + uint8 MIN_SIZE = 1
    + uint8 MAX_SIZE = 128
    ```
- `SequenceOfCartesianPosition3d.msg`
    ```diff
    - int64 MIN_SIZE = 1
    - int64 MAX_SIZE = 16
    + uint8 MIN_SIZE = 1
    + uint8 MAX_SIZE = 16
    ```
- `SequenceOfIdentifier1B.msg`
    ```diff
    - int64 MIN_SIZE = 1
    - int64 MAX_SIZE = 128
    + uint8 MIN_SIZE = 1
    + uint8 MAX_SIZE = 128
    ```
- `TrailerDataSet.msg`
    ```diff
    - int64 MIN_SIZE = 1
    - int64 MAX_SIZE = 8
    + uint8 MIN_SIZE = 1
    + uint8 MAX_SIZE = 8
    ```
- `VruClusterInformation.msg`
    ```diff
    - int64 CLUSTER_BOUNDING_BOX_SHAPE_MIN = 0
    ```
- `WrappedCpmContainer.msg`
    ```diff
    - CpmContainers container_data
    
    + OriginatingVehicleContainer container_data_originating_vehicle_container
    + uint8 CHOICE_CONTAINER_DATA_ORIGINATING_VEHICLE_CONTAINER = 1
    
    + OriginatingRsuContainer container_data_originating_rsu_container
    + uint8 CHOICE_CONTAINER_DATA_ORIGINATING_RSU_CONTAINER = 2
    
    + SensorInformationContainer container_data_sensor_information_container
    + uint8 CHOICE_CONTAINER_DATA_SENSOR_INFORMATION_CONTAINER = 3
    
    + PerceptionRegionContainer container_data_perception_region_container
    + uint8 CHOICE_CONTAINER_DATA_PERCEPTION_REGION_CONTAINER = 4
    
    + PerceivedObjectContainer container_data_perceived_object_container
    + uint8 CHOICE_CONTAINER_DATA_PERCEIVED_OBJECT_CONTAINER = 5
    ```

</details>

### CODE REVIEW

File: .github/workflows/codegen.yml
Comment (lreiher) [inline .github/workflows/codegen.yml:1]: Remove Docker image build from CI?

File: utils/codegen/codegen-py/asn1CodeGenerationUtils.py [hunk 1]
```diff
@@ -339,7 +385,7 @@ def checkTypeMembersInAsn1(asn1_types: Dict[str, Dict]):
 
             # check if type is known
             if member["type"] not in known_types:
-                if ".&" in member["type"]:
+                if ".&" in member["type"]: # TODO: remove
```
Comment (lreiher) [inline utils/codegen/codegen-py/asn1CodeGenerationUtils.py:388]: ?

File: utils/codegen/codegen-py/asn1CodeGenerationUtils.py [hunk 2]
```diff
@@ -551,14 +599,18 @@ def asn1TypeToJinjaContext(asn1_type_name: str, asn1_type_info: Dict, asn1_types
             "is_choice_var": True
         })
 
+        if "identified_by" in asn1_type_info:
```
Comment (lreiher) [inline utils/codegen/codegen-py/asn1CodeGenerationUtils.py:602]:
```suggestion
        # disable choice identifier, if identified by another property (e.g., in the case of CLASS)
        if "identified_by" in asn1_type_info:
```

File: utils/codegen/codegen-py/asn1CodeGenerationUtils.py [hunk 3]
```diff
@@ -551,14 +599,18 @@ def asn1TypeToJinjaContext(asn1_type_name: str, asn1_type_info: Dict, asn1_types
...
+            name = asn1_type_info["identified_by"]
+            context["members"][0]["custom_identifier"] = True
```
Comment (lreiher) [inline utils/codegen/codegen-py/asn1CodeGenerationUtils.py:604]: I don't understand why 0 here is working.

Comment (lreiher) [inline utils/codegen/codegen-py/asn1CodeGenerationUtils.py:604]: Rename `custom_identifier` to `disabled`?

File: utils/codegen/codegen-py/asn1CodeGenerationUtils.py [hunk 4]
```diff
@@ -709,6 +761,30 @@ def asn1TypeToJinjaContext(asn1_type_name: str, asn1_type_info: Dict, asn1_types
...
+        asn1_class = asn1_classes[asn1_type_type.split(".")[0]]
+        for member in asn1_class["members"]:
+            if member["name"] == asn1_type_type.split(".")[1]:
+                class_member = {
+                    "name": asn1_type_info["name"],
+                    "type": member["type"]
+                }
+                if member["type"] not in asn1_types:
+                    class_member["type"] = "CHOICE"
+                    class_member["identified_by"] = "container_id.value" # TODO: get from table
```
Comment (lreiher) [inline utils/codegen/codegen-py/asn1CodeGenerationUtils.py:776]: to be generalized?

File: utils/codegen/codegen-py/asn1CodeGenerationUtils.py [hunk 5]
```diff
@@ -709,6 +761,30 @@ def asn1TypeToJinjaContext(asn1_type_name: str, asn1_type_info: Dict, asn1_types
...
+                    class_member["members"] = []
+                    for value in asn1_values:
+                        for member in asn1_class["members"]:
+                            if member["type"] == asn1_values[value]["type"]:
+                                class_member["members"].append({
+                                    "name": validRosType(value), # TODO: misuse of validRosType; just want to start with a capital letter
+                                    "type": validRosType(value) # TODO: misuse of validRosType; just want to start with a capital letter
```
Comment (lreiher) [inline utils/codegen/codegen-py/asn1CodeGenerationUtils.py:783]: Where is this used and hwy is capital letter needed? If it's only about the capital letter, use `value.capitalize()`?

File: etsi_its_msgs/etsi_its_cpm_ts_msgs/msg/PerceivedObject.msg [hunk 6]
```diff
@@ -126,8 +125,6 @@ bool object_dimension_x_is_present
 
 DeltaTimeMilliSecondSigned object_age
 bool object_age_is_present
-uint16 OBJECT_AGE_MIN = 0
-uint16 OBJECT_AGE_MAX = 2047
```
Comment (lreiher) [inline etsi_its_msgs/etsi_its_cpm_ts_msgs/msg/PerceivedObject.msg:130]: Bug must be fixed before merge, this is an important message.

File: utils/codegen/codegen-py/asn1CodeGenerationUtils.py (hunk 3 above)
Comment (jpbusch) [reply utils/codegen/codegen-py/asn1CodeGenerationUtils.py:604]: `context["members"]` is a list of dicts, but holds only one dict.

File: utils/codegen/codegen-py/asn1CodeGenerationUtils.py (hunk 5 above)
Comment (jpbusch) [reply utils/codegen/codegen-py/asn1CodeGenerationUtils.py:804]: `capitalize()` doesn't work. `"myString".capitalize() -> "Mystring"`

File: .github/workflows/codegen.yml
Comment (jpbusch) [reply .github/workflows/codegen.yml:1]: I have removed the CI job dependencies, but kept the image build.

File: etsi_its_msgs/etsi_its_cpm_ts_msgs/msg/WrappedCpmContainer.msg [hunk 7]
```diff
@@ -26,21 +25,37 @@
...
+uint8 CHOICE_CONTAINER_DATA_ORIGINATING_RSU_CONTAINER = 1
+
+SensorInformationContainer container_data_sensor_information_container
+uint8 CHOICE_CONTAINER_DATA_SENSOR_INFORMATION_CONTAINER = 2
+
+PerceptionRegionContainer container_data_perception_region_container
+uint8 CHOICE_CONTAINER_DATA_PERCEPTION_REGION_CONTAINER = 3
+
+PerceivedObjectContainer container_data_perceived_object_container
+uint8 CHOICE_CONTAINER_DATA_PERCEIVED_OBJECT_CONTAINER = 4
```
Comment (lreiher) [inline etsi_its_msgs/etsi_its_cpm_ts_msgs/msg/WrappedCpmContainer.msg:60]: Choice IDs are not set correctly, leading to erroneous encoding/decoding in conversion.

**Review by lreiher — APPROVED** [review]

**Pull request merged**